lua/entities/gmod_wire_coldetector.lua

```diff
@@ -0,0 +1,227 @@
+	if GetConVarNumber("wire_col_detector_stickprops") == 0 and not gamemode.Call("CanProperty",self:GetPlayer(),"weld",data.HitEntity) then return end
+	
+	if self.StickToWorld == 0 and data.HitEntity:IsWorld() then return end
+	if self.StickToNPC == 0 and (data.HitEntity:IsNPC() or data.HitEntity:IsPlayer()) then return end
+	
+	local ent2pos = data.HitEntity:GetPos()
+	local ent2ang = data.HitEntity:GetAngles()
+	
+	timer.Simple(0,function()
+		if IsValid(ent) and (IsValid(data.HitEntity) or data.HitEntity:IsWorld()) and not IsValid(self.Stick) and  not IsValid(ent:GetParent()) then
+			local bonehit = 0
+			self.StickTo = data.HitEntity
+			
+			angle.pitch = ent2ang.pitch-angle.pitch
+			angle.yaw = ent2ang.yaw-angle.yaw
+			angle.roll = ent2ang.roll-angle.roll
+			vec:Rotate(angle)
+			
+			for i = 0,data.HitEntity:GetPhysicsObjectCount()-1 do
+				if data.HitEntity:GetPhysicsObjectNum(i) == data.HitObject then
+					bonehit = i
+					break
+				end
+			end
+			
+			local newentpos = data.HitEntity:GetPos()
+			local newentang = data.HitEntity:GetAngles()
+			
+			data.HitEntity:SetPos(ent2pos)
+			data.HitEntity:SetAngles(ent2ang)
+			ent:GetPhysicsObject():SetAngles(anglesxd)
+			ent:GetPhysicsObject():SetPos(posxd)
+			ent:SetAngles(anglesxd)
+			ent:SetPos(posxd)
+			
+			if util.IsValidPhysicsObject( data.HitEntity, 0) && (not self.NoPhysics || data.HitEntity:IsWorld()) then
+				self.Stick = constraint.Weld(ent, data.HitEntity,0,bonehit,0,true)
+				if data.HitEntity:IsWorld() then
+					ent:GetPhysicsObject():EnableMotion(false)
+					ent:GetPhysicsObject():Sleep()
+				end
+			else
+				ent:PhysicsDestroy()
+				ent:SetSolid(SOLID_NONE)
+				ent:FollowBone(data.HitEntity,bonehit)
+				
+			end
+			
+			data.HitEntity:SetPos(newentpos)
+			data.HitEntity:SetAngles(newentang)
+		end
+	end)
+end
+
+function ENT:RemoveLink()
+	self.StickTo = nil
+	if IsValid(self.Stick) then
+		self.Stick:Remove()
+	end
+	
+	if IsValid(self.LinkedEnt) then
+		if IsValid(self.LinkedEnt:GetParent()) then
+			self.LinkedEnt:SetParent(nil)
+			self.LinkedEnt:PhysicsInit(SOLID_VPHYSICS)
+			self.LinkedEnt:SetSolid( SOLID_VPHYSICS )
+			timer.Simple(0,function() 
+				if IsValid(self.LinkedEnt) then
+					self.LinkedEnt:GetPhysicsObject():Wake() 
+				end
+			end)
+		end
+	end
+	
+end
+
+function ENT:TriggerInput( name, value )
+	if name == "Stick to world" then
+		self.StickToWorld = value
+		if value == 0 and self.StickTo and self.StickTo:IsWorld() then
+			self:RemoveLink()
+		end
+	elseif name == "Stick to props" then
+		self.StickToProps = value
+		if value == 0 and self.StickTo and not self.StickTo:IsNPC() and not self.StickTo:IsPlayer() and not self.StickTo:IsWorld() then
+			self:RemoveLink()
+		end
+	elseif name == "Stick to npc/players" then
+		self.StickToNPC=value
+		if value == 0 and self.StickTo and (self.StickTo:IsNPC() or self.StickTo:IsPlayer()) then
```

Comment:
This can be shortcut:
````
if value ~= 0 then return end
if not self.StickTo then return end

if self.StickTo:IsNPC() or self.StickTo:IsPlayer() then
[...]
````